clients/unshaded/src/main/java/tachyon/client/file/FileSystemMasterClient.java

```diff
@@ -63,8 +61,19 @@ protected String getServiceName() {
   }
 
   @Override
-  protected void afterConnect() {
+  protected void afterConnect() throws IOException {
     mClient = new FileSystemMasterService.Client(mProtocol);
+    if (mServiceVersion == Constants.UNKNOWN_SERVICE_VERSION) {
+      try {
+        mServiceVersion = mClient.getServiceVersion();
```

Comment:
I think all of these blocks in each of the clients can be factored out, by making all services extend a base service that defines the getServiceVersion() RPC. After that, the clients will also extend from a base class, which would have the getServiceVersion() method.

Here is more information about extending services:
http://stackoverflow.com/questions/21245987/defining-a-service-across-several-files-in-apache-thrift